pep-XXXX.rst

```diff
@@ -0,0 +1,258 @@
+versions of certain core libraries like ``glibc``.
+
+
+External Shared Libraries
+-------------------------
+
+Most desktop and server linux distributions come with a system package manager
+(examples include ``APT`` on Debian-based systems, ``yum`` on
+``RPM``-based systems, and ``pacman`` on Arch linux) that manages, among other
+responsibilities, the installation of shared libraries installed to system
+directories such as ``/usr/lib``. Most non-trivial Python extensions will depend
+on one or more of these shared libraries, and thus function properly only on
+systems where the user has the proper libraries (and the proper
+versions thereof), either installed using their package manager, or installed
+manually by setting certain environment variables such as ``LD_LIBRARY_PATH``
+to notify the runtime linker of the location of the depended-upon shared
+libraries.
+
+
+Versioning of Core Shared Libraries
+-----------------------------------
+
+Even if author or maintainers of a Python extension module with to use no
+external shared libraries, the modules will generally have a dynamic runtime
+dependency on the GNU C library, ``glibc``. While it is possible, statically
+linking ``glibc`` is usually a bad idea because of bloat, and because certain
+important C functions like ``dlopen()`` cannot be called from code that
+statically links ``glibc``. A runtime shared library dependency on a
+system-provided ``glibc`` is unavoidable in practice.
+
+The maintainers of the GNU C library follow a strict symbol versioning scheme
+for backward compatibility. This ensures that binaries compiled against an older
+version of ``glibc`` can run on systems that have a newer ``glibc``. The
+opposite is generally not true -- binaries compiled on newer Linux
+distributions tend to rely upon versioned functions in glibc that are not
+available on older systems.
+
+This generally prevents built distributions compiled on the latest Linux
+distributions from being portable.
+
+
+The ``manylinux_1`` policy
+==========================
+
+For these reasons, to achieve broad portability, Python wheels
+
+ * should depend only on an extremely limited set of external shared
+   libraries; and
+ * should depend only on ``old`` symbol versions in those external shared
+   libraries.
+
+The ``manylinux_1`` policy thus encompasses a standard for what the
+permitted external shared libraries a wheel may depend on, and the maximum
+depended-upon symbol versions therein.
+
+The permitted external shared libraries are: ::
+
+  "libpanelw.so.5", "libncursesw.so.5", "libgcc_s.so.1", "libstdc++.so.6",
+  "libm.so.6", "libdl.so.2", "librt.so.1", "libcrypt.so.1", "libc.so.6",
+  "libnsl.so.1", "libutil.so.1", "libpthread.so.0", "libX11.so.6",
+  "libXext.so.6", "libXrender.so.1", "libICE.so.6", "libSM.so.6",
+  "libGL.so.1", "libgobject-2.0.so.0", "libgthread-2.0.so.0",
+  "libglib-2.0.so.0", "libgmodule-2.0.so.0", and "libgio-2.0.so.0".
```

Comment:
Re: libgmodule: yeah -- looking at the full list in canopy-further-investigation it appears to be used only by wx-gtk (which also depends on a bunch of even heavier stuff like gtk), and there's lots of other stuff on that list that we don't have in the policy (gtk itself, libpng, GLU...). So my inclination is to [draw a simple, defensible, objective line line](https://groups.google.com/d/msg/manylinux-discuss/NR-TN_g5gJk/U7mCeShhFgAJ) rather than try to come up with exactly the right answer for every library? But I could be argued into gmodule, I guess...